### PR TITLE
aws.cloudtrail | check for shadow and organizationtrail

### PR DESCRIPTION
**Pull request by PratMis**
Adding check for shadow and organization trails

### CODE REVIEW

File: c7n/resources/cloudtrail.py [hunk 1]
```diff
@@ -233,8 +233,13 @@ class DeleteTrail(BaseAction):
...
         client = local_session(self.manager.session_factory).client('cloudtrail')
+        shadow_check = IsShadow({'state': True}, self.manager)
+        shadow_check.embedded = True
+        resources = shadow_check.process(resources)
         for r in resources:
-            try:
-                client.delete_trail(Name=r['Name'])
-            except client.exceptions.TrailNotFoundException:
-                continue
+            if not r.get('IsOrganizationTrail', False):
```
Comment (kapilt) [inline c7n/resources/cloudtrail.py]: this is redundant given the shadow check

File: c7n/resources/cloudtrail.py [hunk 2]
```diff
@@ -233,8 +233,13 @@ class DeleteTrail(BaseAction):
 
     def process(self, resources):
         client = local_session(self.manager.session_factory).client('cloudtrail')
+        shadow_check = IsShadow({'state': True}, self.manager)
```
Comment (kapilt) [inline c7n/resources/cloudtrail.py]: you want state: false...

**Comment by PratMis** [comment]
Got it, wasn't aware of shadow trails. I have put in all the fixes.

**Comment by PratMis** [comment]
Looks like there's some problem with publishing codecov results.

File: c7n/resources/cloudtrail.py (hunk 1 above)
Comment (PratMis) [reply c7n/resources/cloudtrail.py]: Looks like we might want to have this check. is_shadow function can return true even if IsOrganizationTrail is false. There's one more condition:
```
        if t.get('IsMultiRegionTrail') and t['HomeRegion'] != self.manager.config.region:
            return True
```

Comment (PratMis) [reply c7n/resources/cloudtrail.py]: Not sure if they are mutually exclusive

Comment (kapilt) [reply c7n/resources/cloudtrail.py]: a shadow trail (either org or multi region from different region) can not be deleted.

**Pull request merged**